fix(ui): checklist parent overflow

ayushsharma74 opened the pull request:
# Description
Fixes #614 
Fixed overflow of note component when checklist content is too long

## Before
https://github.com/user-attachments/assets/8ff6b585-a661-4a8b-8e65-1a3c9b07a24b

## After
https://github.com/user-attachments/assets/9d26157d-c9ef-442b-bedd-19b9c7d01390

ayushsharma74 commented:
Hello @slavingia, the tests are now passing successfully. I would appreciate it if you could reopen this PR and review it at your convenience.

ayushsharma74 commented on app/boards/[id]/page.tsx:
```diff
@@ -251,6 +252,18 @@ export default function BoardPage({ params }: { params: Promise<{ id: string }>
 
+  const handleNoteHeightChange = useCallback((noteId: string, newHeight: number) => {
+    setNoteHeights((prev) => {
+      if (prev[noteId] === newHeight) {
```
If there is no change in height, just return the previous height to prevent unnecessary re-renders

ayushsharma74 commented on components/note.tsx:
```diff
@@ -334,8 +336,21 @@ export function Note({
 
+  const noteRef = React.useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
```
This introduces a feedback loop for note layout. Each note’s container is measured using a ref and useLayoutEffect to capture its offsetHeight before paint. The height is compared against the previous value to avoid redundant updates, and if it changes, onHeightChange notifies the parent to re-calculate the grid so notes reposition correctly as their content grows or shrinks.

ayushsharma74 commented:
> zero comments to help me understand code changes, no before video....

Apologies for the earlier oversight, I’ve added comments and before/after videos. Could you kindly review the PR when you have a moment?

ayushsharma74 commented:
hi @slavingia Sorry for the ping again, all tests are passing, could you please review?

slavingia commented on app/boards/[id]/page.tsx:
```diff
@@ -895,14 +909,14 @@ export default function BoardPage({ params }: { params: Promise<{ id: string }>
               onArchive={boardId !== "archive" ? handleArchiveNote : undefined}
               onUnarchive={boardId === "archive" ? handleUnarchiveNote : undefined}
               onCopy={handleCopyNote}
+              onHeightChange={handleNoteHeightChange}
```
shouldn't be on Height Change right, just on content change should always re-render? I think pursuing a CSS only approach would be beneficial. Similar to homepage

ayushsharma74 commented:
Refactored the current notes component to "only CSS" approach and removed all the complex calculations to measure its height and width

Pushing e2e tests soon 
 
# Desktop
https://github.com/user-attachments/assets/10f24be4-e5ac-494b-adf0-b220f83a82ca

# Mobile
https://github.com/user-attachments/assets/8b6aaa0b-3e1a-4918-b025-30ff1e8e2312

ayushsharma74 commented on components/checklist-item.tsx:
```diff
@@ -55,14 +55,14 @@ export function ChecklistItem({
   };
 
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
```
using useLayoutEffect here to prevent flickering

slavingia commented on app/boards/[id]/page.tsx:
```diff
@@ -885,40 +835,38 @@ export default function BoardPage({ params }: { params: Promise<{ id: string }>
-          minHeight: "calc(100vh - 64px)", // Account for header height
-        }}
-      >
+      <div ref={boardRef} className="relative w-full p-4">
```
Why the additional padding?

ayushsharma74 replied:
this padding is for adding outer margin to note, `m-4` also works the same 

without it notes are sticking to the corners like this
<img width="700" height="700" alt="localhost_3000_boards_cmehag5b40007dstkv6lvqbdi" src="https://github.com/user-attachments/assets/58a36094-4245-406c-8fe1-e5638992a802" />

slavingia commented on app/boards/[id]/page.tsx:
```diff
@@ -885,40 +835,38 @@ export default function BoardPage({ params }: { params: Promise<{ id: string }>
+                  onUnarchive={boardId === "archive" ? handleUnarchiveNote : undefined}
+                  onCopy={handleCopyNote}
+                  showBoardName={boardId === "all-notes" || boardId === "archive"}
+                  className="shadow-md shadow-black/10 w-full h-full p-4"
```
The shadow and padding should be on the component

ayushsharma74 replied:
Got it! I’ll move the shadow and padding directly onto the component.

slavingia commented on app/public/boards/[id]/page.tsx:
```diff
@@ -1,31 +1,26 @@
 import Link from "next/link";
 import { BetaBadge } from "@/components/ui/beta-badge";
 import { FullPageLoader } from "@/components/ui/loader";
+
```
Why?

laugardie commented:
@ayushsharma74 made a few small tweaks. Matched the padding between the topbar and cards, and fixed a couple of minor UI issues I noticed. Everything else looks good!

slavingia commented:
These seem like bugs. Are they happening for you on prod?

<img width="673" height="753" alt="image" src="https://github.com/user-attachments/assets/f0a9c095-04ab-43d5-9674-ecad62904561" />

ayushsharma74 commented:
I noticed a difference between local and gumboard.com . Locally, when I create a new note in dark mode locally it uses the light mode colors, but on production (in dark mode) it applies the dark colors. I’m not sure what’s causing this discrepancy.

slavingia commented on app/boards/[id]/page.tsx:
```diff
@@ -885,40 +836,38 @@ export default function BoardPage({ params }: { params: Promise<{ id: string }>
+                  showBoardName={boardId === "all-notes" || boardId === "archive"}
+                  className="w-full h-full"
+                  style={{
+                    backgroundColor: resolvedTheme === "dark" ? "#18181B" : note.color,
```
we can't really merge this until you resolve this on your local, as your code does update this and it may lead to ongoing issues.

slavingia commented on app/public/boards/[id]/page.tsx:
```diff
@@ -1,6 +1,6 @@
 "use client";
 
-import { useState, useEffect, useRef, useMemo } from "react";
```
can you reflect laura's design changes in this file?

slavingia commented:
Likely will be some merge conflicts that should be resolved as we merged in a large boards layout refactor.

ayushsharma74 commented:
@slavingia  I looked into the card color issue in dark mode. Locally, switching to dark mode doesn’t update the html class to dark, so the colors don’t change in only notes for some reason. However, after deploying to Vercel, it works correctly as shown in the video.

I’m not sure, but this seems to be related to `next-themes` rather than our code. Curious to hear your thoughts on where this might be coming from and whether we should fix the implementation to handle the local issue as well?

https://github.com/user-attachments/assets/5c8c62fa-0191-4aa6-880e-af95896aa3b6

ayushsharma74 commented:
Found some optimization issues in stress testing, fixing them and will open a new PR.

ayushsharma74 commented:
@kr-sushil ik, i was referring to the issues in animation not the overflow.
